tests/unit/test_charm.py

```diff
@@ -606,6 +606,8 @@ def test_exec_error_on_client_created_event_emitted(
     caplog.set_level(logging.ERROR)
     harness.set_can_connect(CONTAINER_NAME, True)
     setup_peer_relation(harness)
+    # needed otherwise self.tracing.is_ready logs an error
+    setup_tempo_relation(harness)
```

Comment:
I think we should just change the check we do on the logs, something like this should work:
```python
    assert f"Exited with code: {err.exit_code}. Stderr: {err.stderr}" in caplog.record_tuples[0]
```